api/v1alpha1/common_types.go

```diff
@@ -37,4 +39,16 @@ type ControlPlaneEndpointSpec struct {
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=65535
 	Port int32 `json:"port"`
+
+	// Configuration for the virtual IP provider.
+	// +optional
+	VirtualIPSpec *ControlPlaneVirtualIPSpec `json:"virtualIP,omitempty"`
```

Comment:
wdyt if we use the `*corev1.LocalObjectReference ` object reference here and default it to the kubevip configmap reference that we use in the chart? I think it could make the schema generic and the handler can use virtual ip specific logic at runtime. 
This way we dont have to provide name of the configmap as arg to CAREN server.

PS: I understand you just changed the schema to include `Provider` string.

Reply:
I thought about this at first but wanted to keep it similar to the addons.
If its part of the API it would be pretty easy to end up with different clusters with different config over a period of time since it would no longer come from a global config.

If we do prefer that approach though, IMO it should be a more general change across the whole API.

Reply:
ok. So we want that a CAREN instance can only support one VirtualIP Provider and all the clusters will use the same provider. The CAREN client has to make sure that they use correct value for `Provider` Enum that matches with what configmap they provided when deploying CAREN. Is that well established contract?

Reply:
Ahh I think I understand where the original question is coming from. When we support additional providers there would be [more flags](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/558/files#diff-7a859c2cff68db150a13157650eadaa81d36ff680116ed54d910acad6fe8ab6eR62-R63) and charts/cluster-api-runtime-extensions-nutanix/templates/kube-vip/manifests/kube-vip-configmap.yaml deployed for the new providers.
So the user would have an option on which provider to deploy for different clusters, but what the config for that specific provider will be the same across all clusters.

Reply:
Let me change up the chart flags to make that a little more clear.

Reply:
Thanks for making it clear. Flags for provider would work.